gum/arch-arm64/gumarm64writer.c

```diff
@@ -707,6 +707,29 @@ gum_arm64_writer_put_add_reg_reg_imm (GumArm64Writer * self,
 }
 
 void
+gum_arm64_writer_put_add_reg_reg_reg (GumArm64Writer * self,
+                                      GumArm64Reg dst_reg,
+                                      GumArm64Reg left_reg,
+                                      GumArm64Reg right_reg)
+{
+  GumArm64RegInfo rd, rl, rr;
+
+  gum_arm64_writer_describe_reg (self, dst_reg, &rd);
+  gum_arm64_writer_describe_reg (self, left_reg, &rl);
+  gum_arm64_writer_describe_reg (self, right_reg, &rr);
+
+  g_assert_cmpuint (rd.width, ==, rl.width);
+  g_assert_cmpuint (rd.width, ==, rr.width);
+
+  guint32 flags = 0;
+  if(rd.width == 64) {
```

Comment:
Two nitpicks: should be space between `if` and `(`, and curly braces should align, though in this case they should be omitted to make the code more readable.